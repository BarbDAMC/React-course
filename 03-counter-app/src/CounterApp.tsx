import { useState } from 'react';
import './styles.css';

interface CounterAppProps {
  value: number
}

export const CounterApp = ({ value } : CounterAppProps) => {
  const [ counter, setCounter ] = useState(value);

  const handleReset = () => setCounter(value);
  const handleAdd = () => setCounter(counter + 1);
  const handleSubtract = () => setCounter(counter - 1);
  

  return (
    <>
      <h1>CounterApp</h1>
      <h2> { counter } </h2>

      <button onClick={handleAdd}> +1</button>
      <button onClick={handleSubtract}> -1</button>
      <button onClick={handleReset}>Reset</button>
    </>
  )
}
