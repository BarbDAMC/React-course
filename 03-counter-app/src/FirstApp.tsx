import './styles.css'

interface FirstAppProps {
  title: string,
  subtitle: string,
  name: string
}


export const FirstApp = ({title, subtitle, name} : FirstAppProps) => {
  return (
     <>
      <h1>{title}</h1>
      <p>{ subtitle }</p>
      <p>{ name }</p>
    </>  
  )
}




