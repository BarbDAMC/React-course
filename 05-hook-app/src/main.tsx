import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
// import { HooksApp } from './HooksApp'
// import { CounterApp } from './01-useState/CounterApp'
// import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'
// import { SimpleForm } from './02-useEffect/SimpleForm'
// import { SimpleFormWithCustomHook } from './02-useEffect/SimpleFormWithCustomHook'
// import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks';
// import { FocusScreen } from './04-useRef/FocusScreen';
// import { Layout } from './05-useLayoutEffect/Layout'
// import { Memorize } from './06-useMemo/Memorize'
// import { MemoHook } from './06-useMemo/MemoHook'
// import { CallbackHook } from './06-useMemo/CallbackHook'
import './index.css'
import { Padre } from './07-tarea-memo/Padre'


createRoot(document.getElementById('root')!).render(
  <StrictMode>
    {/* <HooksApp /> */}
    {/* <CounterApp /> */}
    {/* <CounterWithCustomHook /> */}
    {/* <SimpleForm /> */}
    {/* <SimpleFormWithCustomHook /> */}
    {/* <MultipleCustomHooks /> */}
    {/* <FocusScreen /> */}
    {/* <Layout /> */}
    {/* <Memorize /> */}
    {/* <MemoHook /> */}
    {/* <CallbackHook /> */}
    <Padre />
  </StrictMode>,
)
