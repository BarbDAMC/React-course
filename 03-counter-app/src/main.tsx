import React from 'react';
import ReactDOM from 'react-dom/client';
// import App from './App';
// import { FirstApp } from './FirstApp';
import { CounterApp } from './CounterApp';

const root = document.getElementById('root');
ReactDOM.createRoot(root!).render(
  <React.StrictMode>
    {/* <App /> */}
    {/* <FirstApp 
      title="Hola, soy Goku" 
      subtitle='123'
      name="Barbara"
    /> */}

    <CounterApp value={0} />
  </React.StrictMode>
)
