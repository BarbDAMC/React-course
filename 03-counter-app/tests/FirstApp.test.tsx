import { render, screen } from '@testing-library/react'
import { FirstApp } from '../src/FirstApp';

describe('<FirstApp />', () => {
  const title = 'Hola';
  const subtitle = 'mundo';
  const name = 'Barbara';

  test('should match with the snapshot', () => {
    const { container } = render(<FirstApp title={title} subtitle={subtitle} name={name} />)
    expect( container ).toMatchSnapshot();
  })

  test('should show the title', () => {
    render(<FirstApp title={title} subtitle={subtitle} name={name} />);
    
    expect( screen.getByText(title) ).toBeTruthy();
    expect( screen.getByRole('heading', {level: 1}).innerHTML ).toContain(title);
  })

  test('should show the subtitle', () => {
    render(<FirstApp title={title} subtitle={subtitle} name={name} />);
    
    expect( screen.getAllByText(subtitle).length ).toBe(2);
  })
})
