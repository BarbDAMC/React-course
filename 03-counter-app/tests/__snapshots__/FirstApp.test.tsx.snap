// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<FirstApp /> should match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola
  </h1>
  <p>
    mundo
  </p>
  <p>
    mundo
  </p>
  <p>
    Barbara
  </p>
</div>
`;
